Sources/Algorithms/Stride.swift

```diff
@@ -88,12 +102,36 @@ extension Stride: Sequence {
   }
   
   @inlinable
-  public func makeIterator() -> Stride<Base>.Iterator {
+  public func makeIterator() -> Iterator {
     Iterator(iterator: base.makeIterator(), stride: stride)
   }
 }
 
-extension Stride: Collection where Base: Collection {
+/// A wrapper that strides over a base sequence or collection.
+public struct Stride<Base: Collection> {
+  @usableFromInline
+  internal let base: Base
+  
+  @usableFromInline
+  internal let stride: Int
+  
+  @usableFromInline
+  internal init(base: Base, stride: Int) {
+    precondition(stride > 0, "striding must be greater than zero")
+    self.base = base
+    self.stride = stride
+  }
+}
+
+extension Stride {
+  @inlinable
+  public func striding(by step: Int) -> Self {
+    Stride(base: base, stride: stride * step)
+  }
+}
+
+extension Stride: Sequence & Collection {
```

Comment:
```suggestion
extension Stride: Sequence, Collection {
```

Nit: This is the more idiomatic spelling.

Reply:
Or simply `extension Stride: Collection`, as this also implies `Sequence` 🙂